### DISCUSSION TITLE
编译源代码报错，求救。。。

### DISCUSSION THREAD

**Discussion by gisIdea** [apache/doris#37501] [A - General / Q&A]
win64电脑 dockerdesktop里 下载的apache/doris镜像 然后进去docker 编译的是2.0的源代码版本 
先下载了第三方依赖 并安装了第三方依赖包 然后就开始到src目录build 报错是说
/opt/apache-doris-2.0.0-src/be/src/olap/rowset/segment_v2/inverted_index_reader.h:183:10: error: virtual function 'visit' has a different return type ('void') than the function it overrides (which has return type 'int')

这咋搞？
visit这个方法的返回值不应该是void应该是int？我不能去修改源代码啊？是不是版本问题？要卸载那个第三方依赖包并重新安装呢？应该都用啥版本呢？新手求教~求大佬不吝赐祭 谢谢~

build完整日志如下：

[root@04fc7fd63855 apache-doris-2.0.0-src]# sh ./build.sh
Python 2.7.5
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /usr/share/maven
Java version: 1.8.0_412, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.412.b08-1.el7_9.x86_64/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux", version: "5.15.153.1-microsoft-standard-wsl2", arch: "amd64", family: "unix"
cmake version 3.22.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
ninja 1.8.2
ccache version 4.8
Update apache-orc submodule ...
fatal: Not a git repository (or any of the parent directories): .git
Update apache-orc submodule failed, start to download and extract apache-orc package ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:04 --:--:--     0
100 13.8M    0 13.8M    0     0   657k      0 --:--:--  0:00:21 --:--:-- 1258k
Update clucene submodule ...
fatal: Not a git repository (or any of the parent directories): .git
Update clucene submodule failed, start to download and extract clucene package ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
100 8152k    0 8152k    0     0   393k      0 --:--:--  0:00:20 --:--:--  936k
Get params:
    BUILD_FE                    -- 1
    BUILD_BE                    -- 1
    BUILD_BROKER                -- 1
    BUILD_AUDIT                 -- 1
    BUILD_META_TOOL             -- OFF
    BUILD_INDEX_TOOL            -- OFF
    BUILD_SPARK_DPP             -- 1
    BUILD_BE_JAVA_EXTENSIONS    -- 1
    BUILD_HIVE_UDF              -- 1
    PARALLEL                    -- 3
    CLEAN                       -- 0
    WITH_MYSQL                  -- OFF
    WITH_LZO                    -- OFF
    GLIBC_COMPATIBILITY         -- ON
    USE_AVX2                    -- ON
    USE_LIBCPP                  -- OFF
    USE_DWARF                   -- OFF
    USE_UNWIND                  -- ON
    STRIP_DEBUG_INFO            -- OFF
    USE_MEM_TRACKER             -- ON
    USE_JEMALLOC                -- ON
    USE_BTHREAD_SCANNER         -- OFF
    ENABLE_STACKTRACE           -- ON
    DENABLE_CLANG_COVERAGE      -- OFF
    DISPLAY_BUILD_TIME          -- OFF
    ENABLE_PCH                  -- ON

Python 2.7.5
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /usr/share/maven
Java version: 1.8.0_412, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.412.b08-1.el7_9.x86_64/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux", version: "5.15.153.1-microsoft-standard-wsl2", arch: "amd64", family: "unix"
cmake version 3.22.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
ninja 1.8.2
ccache version 4.8
Build generated code
make -C script
make[1]: Entering directory `/opt/apache-doris-2.0.0-src/gensrc/script'
/opt/apache-doris-2.0.0-src/gensrc/script/gen_build_version.sh
make[1]: Leaving directory `/opt/apache-doris-2.0.0-src/gensrc/script'
make -C proto
make[1]: Entering directory `/opt/apache-doris-2.0.0-src/gensrc/proto'
make[1]: Nothing to be done for `all'.
make[1]: Leaving directory `/opt/apache-doris-2.0.0-src/gensrc/proto'
make -C thrift
make[1]: Entering directory `/opt/apache-doris-2.0.0-src/gensrc/thrift'
make[1]: Nothing to be done for `all'.
make[1]: Leaving directory `/opt/apache-doris-2.0.0-src/gensrc/thrift'
Done
Build Backend: Release
-- Make program: /var/local/ldb-toolchain/bin/ninja
-- Use ccache: -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
-- Extra cxx flags:
-- Build fs benchmark tool: OFF
-- GLIBC_COMPATIBILITY is ON
-- USE_LIBCPP is OFF
-- USE_MEM_TRACKER is ON
-- USE_JEMALLOC is ON
-- USE_UNWIND is ON
-- ENABLE_PCH is ON
-- Build type is RELEASE
-- Build target arch is x86_64
-- THIRDPARTY_DIR is /var/local/thirdparty/installed
-- make test: OFF
-- build fs benchmark tool: OFF
-- build gensrc if necessary
make: Entering directory `/opt/apache-doris-2.0.0-src/gensrc'
make -C script
make[1]: Entering directory `/opt/apache-doris-2.0.0-src/gensrc/script'
/opt/apache-doris-2.0.0-src/gensrc/script/gen_build_version.sh
get java cmd: /usr/lib/jvm/java-1.8.0/bin/java
get java version: openjdk full version \"1.8.0_412-b08\"
make[1]: Leaving directory `/opt/apache-doris-2.0.0-src/gensrc/script'
make -C proto
make[1]: Entering directory `/opt/apache-doris-2.0.0-src/gensrc/proto'
make[1]: Nothing to be done for `all'.
make[1]: Leaving directory `/opt/apache-doris-2.0.0-src/gensrc/proto'
make -C thrift
make[1]: Entering directory `/opt/apache-doris-2.0.0-src/gensrc/thrift'
make[1]: Nothing to be done for `all'.
make[1]: Leaving directory `/opt/apache-doris-2.0.0-src/gensrc/thrift'
make: Leaving directory `/opt/apache-doris-2.0.0-src/gensrc'
-- Found Boost: /var/local/thirdparty/installed/lib64/cmake/Boost-1.81.0/BoostConfig.cmake (found suitable version "1.81.0", minimum required is "1.81.0") found components: system date_time
-- Build type: RELEASE
-- compiler Clang version 16.0.1
-- Disable the metrics collection
-- SNAPPY_HOME: /var/local/thirdparty/installed
-- Found the Snappy header: /var/local/thirdparty/installed/include/snappy.h
-- Found the Snappy library: /var/local/thirdparty/installed/lib/libsnappy.a
-- ZLIB_HOME: /var/local/thirdparty/installed
-- Found the ZLIB header: /var/local/thirdparty/installed/include/zlib.h
-- Found the ZLIB library: /var/local/thirdparty/installed/lib/libz.so
-- Found the ZLIB static library: /var/local/thirdparty/installed/lib/libz.a
-- ZSTD_HOME: /var/local/thirdparty/installed
-- Found the zstd header: /var/local/thirdparty/installed/include/zstd/zstd.h
-- Found the zstd library: /var/local/thirdparty/installed/lib/libzstd.a
-- Found the zstd static library: /var/local/thirdparty/installed/lib/libzstd.a
-- LZ4_HOME: /var/local/thirdparty/installed
-- Found the LZ4 header: /var/local/thirdparty/installed/include/lz4/lz4.h
-- Found the LZ4 library: /var/local/thirdparty/installed/lib/liblz4.a
-- PROTOBUF_HOME: /var/local/thirdparty/installed
-- Found the Protobuf headers: /var/local/thirdparty/installed/include
-- Found the Protobuf library: /var/local/thirdparty/installed/lib/libprotobuf.a
-- Found the Protoc library: /var/local/thirdparty/installed/lib/libprotoc.a
-- Found the Protoc executable: /var/local/thirdparty/installed/bin/protoc
-- Found the Protobuf static library: /var/local/thirdparty/installed/lib/libprotobuf.a
-- Found the Protoc static library: /var/local/thirdparty/installed/lib/libprotoc.a
CMake Deprecation Warning at src/clucene/CMakeLists.txt:20 (CMAKE_MINIMUM_REQUIRED):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

Compiling as RELEASE
-- old Boost_INCLUDE_DIR    : /var/local/thirdparty/installed/include
-- Found Boost: /var/local/thirdparty/installed/lib64/cmake/Boost-1.81.0/BoostConfig.cmake (found version "1.81.0")
Boost found
-- Boost_INCLUDE_DIR    : /var/local/thirdparty/installed/include
-- Looking for MapViewOfFile
-- Looking for MapViewOfFile(0,0,0,0,0) - not found
-- Looking for _close
-- Looking for _close((int)0) - not found
-- Looking for _read
-- Looking for _read((int)0, (void*)0, (unsigned int)0) - not found
-- Looking for _open
-- Looking for _open(0,0,0) - not found
-- Looking for _write
-- Looking for _write((int)0, (const void*)0, (unsigned int)0) - not found
-- Looking for _unlink
-- Looking for _unlink((const char*)0) - not found
-- Looking for _ftime
-- Looking for _ftime(0) - not found
-- Looking for _mkdir
-- Looking for _mkdir((const char*)0) - not found
-- Checking support new float byte<->float conversions
-- Checking support new float byte<->float conversions - yes
-- Linker LLD is available, change linker to LLD
-- Some symbols from glibc will be replaced for compatibility
-- Compiler: Clang-16.0.1
-- CXX Standard: 20
-- C Standard: 17
-- CXX Flags:   -O3 -DNDEBUG
-- C Flags:   -O3 -DNDEBUG
-- CC Flags:
-- Compiler Options: -g;-Wall;-Wextra;-Werror;-pthread;-fstrict-aliasing;-fno-omit-frame-pointer;$<$<COMPILE_LANGUAGE:CXX>:-Wnon-virtual-dtor>;-Wno-unused-parameter;-Wno-sign-compare;-fcolor-diagnostics;-Wpedantic;-Wunused;-Wunused-command-line-argument;-Wunused-exception-parameter;-Wunused-volatile-lvalue;-Wunused-template;-Wunused-member-function;-Wunused-macros;-Wno-vla-extension;-D__STDC_FORMAT_MACROS;-DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG;-DBOOST_SYSTEM_NO_DEPRECATED;-DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX=1;-DBRPC_ENABLE_CPU_PROFILER;-DS2_USE_GFLAGS;-DS2_USE_GLOG;-msse4.2;-mavx2;-DUSE_MEM_TRACKER;-DUSE_JEMALLOC;-DUSE_UNWIND;-gdwarf-aranges;-DENABLE_STACKTRACE
-- Compiler Definitions: BOOST_STACKTRACE_USE_BACKTRACE;HAVE_INTTYPES_H;HAVE_NETINET_IN_H;USE_HADOOP_HDFS
-- Doris Dependencies: -Wl,--start-group;backtrace;rocksdb;cyrus-sasl;libs2;snappy;Boost::date_time;thrift;thriftnb;glog;re2;hyperscan;pprof;lz4;libevent;libevent_pthreads;idn;gsasl;curl;libz;libbz2;gflags;brpc;protobuf;openssl;crypto;leveldb;bitshuffle;roaring;fmt;jemalloc;jemalloc_arrow;brotlicommon;brotlidec;brotlienc;zstd;arrow;parquet;odbc;cctz;minizip;opentelemetry_common;opentelemetry_exporter_zipkin_trace;opentelemetry_resources;opentelemetry_version;opentelemetry_exporter_ostream_span;opentelemetry_trace;opentelemetry_http_client_curl;opentelemetry_exporter_otlp_http;opentelemetry_exporter_otlp_http_client;opentelemetry_otlp_recordable;opentelemetry_proto;aws-sdk-s3;aws-sdk-core;aws-sdk-transfer;aws-checksums;aws-c-io;aws-c-event-stream;aws-c-common;aws-c-cal;aws-s2n;aws-c-s3;aws-c-auth;aws-crt-cpp;aws-c-compression;aws-c-http;aws-c-mqtt;aws-sdk-s3-crt;librdkafka_cpp;librdkafka;xml2;lzma;simdjson;hadoop_hdfs;absl::flat_hash_set;absl::str_format;krb5support;krb5;com_err;gssapi_krb5;k5crypto;libunwind;orc;ic;clucene-core-static;clucene-shared-static;clucene-contribs-lib;-Wl,--end-group
-- Link Flags: -Wl,--start-group;Agent;Common;Exec;Exprs;Gutil;IO;Olap;Rowset;Runtime;Service;Udf;Util;DorisGen;Webserver;Geo;Vec;Pipeline;-Wl,--end-group;-Wl,--start-group;backtrace;rocksdb;cyrus-sasl;libs2;snappy;Boost::date_time;thrift;thriftnb;glog;re2;hyperscan;pprof;lz4;libevent;libevent_pthreads;idn;gsasl;curl;libz;libbz2;gflags;brpc;protobuf;openssl;crypto;leveldb;bitshuffle;roaring;fmt;jemalloc;jemalloc_arrow;brotlicommon;brotlidec;brotlienc;zstd;arrow;parquet;odbc;cctz;minizip;opentelemetry_common;opentelemetry_exporter_zipkin_trace;opentelemetry_resources;opentelemetry_version;opentelemetry_exporter_ostream_span;opentelemetry_trace;opentelemetry_http_client_curl;opentelemetry_exporter_otlp_http;opentelemetry_exporter_otlp_http_client;opentelemetry_otlp_recordable;opentelemetry_proto;aws-sdk-s3;aws-sdk-core;aws-sdk-transfer;aws-checksums;aws-c-io;aws-c-event-stream;aws-c-common;aws-c-cal;aws-s2n;aws-c-s3;aws-c-auth;aws-crt-cpp;aws-c-compression;aws-c-http;aws-c-mqtt;aws-sdk-s3-crt;librdkafka_cpp;librdkafka;xml2;lzma;simdjson;hadoop_hdfs;absl::flat_hash_set;absl::str_format;krb5support;krb5;com_err;gssapi_krb5;k5crypto;libunwind;orc;ic;clucene-core-static;clucene-shared-static;clucene-contribs-lib;-Wl,--end-group;-static-libstdc++;-static-libgcc;-lstdc++fs;-lresolv;jemalloc;glibc-compatibility-explicit;glibc-compatibility;-lrt;-l:libbfd.a;-liberty;-lc;-lm;-ldl;-pthread
-- dir='/opt/apache-doris-2.0.0-src/be/src/apache-orc/c++/include'
-- dir='/opt/apache-doris-2.0.0-src/be/build_Release/src/apache-orc/c++/include'
-- dir='/opt/apache-doris-2.0.0-src/be/build_Release/src/clucene/src/shared'
-- dir='/opt/apache-doris-2.0.0-src/be/src/clucene/src/core'
-- dir='/opt/apache-doris-2.0.0-src/be/src/clucene/src/shared'
-- dir='/opt/apache-doris-2.0.0-src/be/src/clucene/src/contribs-lib'
-- dir='/opt/apache-doris-2.0.0-src/be/src'
-- dir='/opt/apache-doris-2.0.0-src/be/test'
-- dir='/opt/apache-doris-2.0.0-src/be/../gensrc/build'
-- dir='/var/local/thirdparty/installed/include'
-- dir='/var/local/thirdparty/installed/gperftools/include'
-- dir='/usr/lib/jvm/java-1.8.0/include'
-- dir='/usr/lib/jvm/java-1.8.0/include/linux'
-- Configuring done
-- Generating done
-- Build files have been written to: /opt/apache-doris-2.0.0-src/be/build_Release
[0/2] Re-checking globbed directories...
[1/860] Building CXX object src/agent/CMakeFiles/Agent.dir/agent_server.cpp.o
FAILED: src/agent/CMakeFiles/Agent.dir/agent_server.cpp.o
ccache /var/local/ldb-toolchain/bin/clang++ -DBOOST_STACKTRACE_USE_BACKTRACE -DHAVE_INTTYPES_H -DHAVE_NETINET_IN_H -DUSE_HADOOP_HDFS -I/opt/apache-doris-2.0.0-src/be/src/apache-orc/c++/include -I/opt/apache-doris-2.0.0-src/be/build_Release/src/apache-orc/c++/include -I/opt/apache-doris-2.0.0-src/be/build_Release/src/clucene/src/shared -I/opt/apache-doris-2.0.0-src/be/src/clucene/src/core -I/opt/apache-doris-2.0.0-src/be/src/clucene/src/shared -I/opt/apache-doris-2.0.0-src/be/src/clucene/src/contribs-lib -I/opt/apache-doris-2.0.0-src/be/src -I/opt/apache-doris-2.0.0-src/be/test -I/usr/lib/jvm/java-1.8.0/include -I/usr/lib/jvm/java-1.8.0/include/linux -isystem /opt/apache-doris-2.0.0-src/be/../gensrc/build -isystem /var/local/thirdparty/installed/include -isystem /var/local/thirdparty/installed/gperftools/include -O3 -DNDEBUG  -O3 -DNDEBUG   -D OS_LINUX -g -Wall -Wextra -Werror -pthread -fstrict-aliasing -fno-omit-frame-pointer -Wnon-virtual-dtor -Wno-unused-parameter -Wno-sign-compare -fcolor-diagnostics -Wpedantic -Wunused -Wunused-command-line-argument -Wunused-exception-parameter -Wunused-volatile-lvalue -Wunused-template -Wunused-member-function -Wunused-macros -Wno-vla-extension -D__STDC_FORMAT_MACROS -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX=1 -DBRPC_ENABLE_CPU_PROFILER -DS2_USE_GFLAGS -DS2_USE_GLOG -msse4.2 -mavx2 -DUSE_MEM_TRACKER -DUSE_JEMALLOC -DUSE_UNWIND -gdwarf-aranges -DENABLE_STACKTRACE -std=gnu++20 -Winvalid-pch -Xclang -include-pch -Xclang /opt/apache-doris-2.0.0-src/be/build_Release/CMakeFiles/pch.dir/cmake_pch.hxx.pch -Xclang -include -Xclang /opt/apache-doris-2.0.0-src/be/build_Release/CMakeFiles/pch.dir/cmake_pch.hxx -MD -MT src/agent/CMakeFiles/Agent.dir/agent_server.cpp.o -MF src/agent/CMakeFiles/Agent.dir/agent_server.cpp.o.d -o src/agent/CMakeFiles/Agent.dir/agent_server.cpp.o -c /opt/apache-doris-2.0.0-src/be/src/agent/agent_server.cpp
In file included from /opt/apache-doris-2.0.0-src/be/src/agent/agent_server.cpp:30:
In file included from /opt/apache-doris-2.0.0-src/be/src/agent/task_worker_pool.h:35:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/tablet.h:49:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/rowset/rowset_reader.h:26:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/iterators.h:24:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/block_column_predicate.h:32:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/column_predicate.h:24:
/opt/apache-doris-2.0.0-src/be/src/olap/rowset/segment_v2/inverted_index_reader.h:183:10: error: virtual function 'visit' has a different return type ('void') than the function it overrides (which has return type 'int')
    void visit(int row_id, std::vector<uint8_t>& packed_value) override;
    ~~~~ ^
/opt/apache-doris-2.0.0-src/be/src/clucene/src/core/CLucene/util/bkd/bkd_reader.h:65:21: note: overridden virtual function is here
        virtual int visit(int docid, std::vector<uint8_t> &packedValue) = 0;
                ~~~ ^
1 error generated.
[2/860] Building CXX object src/agent/CMakeFiles/Agent.dir/heartbeat_server.cpp.o
FAILED: src/agent/CMakeFiles/Agent.dir/heartbeat_server.cpp.o
ccache /var/local/ldb-toolchain/bin/clang++ -DBOOST_STACKTRACE_USE_BACKTRACE -DHAVE_INTTYPES_H -DHAVE_NETINET_IN_H -DUSE_HADOOP_HDFS -I/opt/apache-doris-2.0.0-src/be/src/apache-orc/c++/include -I/opt/apache-doris-2.0.0-src/be/build_Release/src/apache-orc/c++/include -I/opt/apache-doris-2.0.0-src/be/build_Release/src/clucene/src/shared -I/opt/apache-doris-2.0.0-src/be/src/clucene/src/core -I/opt/apache-doris-2.0.0-src/be/src/clucene/src/shared -I/opt/apache-doris-2.0.0-src/be/src/clucene/src/contribs-lib -I/opt/apache-doris-2.0.0-src/be/src -I/opt/apache-doris-2.0.0-src/be/test -I/usr/lib/jvm/java-1.8.0/include -I/usr/lib/jvm/java-1.8.0/include/linux -isystem /opt/apache-doris-2.0.0-src/be/../gensrc/build -isystem /var/local/thirdparty/installed/include -isystem /var/local/thirdparty/installed/gperftools/include -O3 -DNDEBUG  -O3 -DNDEBUG   -D OS_LINUX -g -Wall -Wextra -Werror -pthread -fstrict-aliasing -fno-omit-frame-pointer -Wnon-virtual-dtor -Wno-unused-parameter -Wno-sign-compare -fcolor-diagnostics -Wpedantic -Wunused -Wunused-command-line-argument -Wunused-exception-parameter -Wunused-volatile-lvalue -Wunused-template -Wunused-member-function -Wunused-macros -Wno-vla-extension -D__STDC_FORMAT_MACROS -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX=1 -DBRPC_ENABLE_CPU_PROFILER -DS2_USE_GFLAGS -DS2_USE_GLOG -msse4.2 -mavx2 -DUSE_MEM_TRACKER -DUSE_JEMALLOC -DUSE_UNWIND -gdwarf-aranges -DENABLE_STACKTRACE -std=gnu++20 -Winvalid-pch -Xclang -include-pch -Xclang /opt/apache-doris-2.0.0-src/be/build_Release/CMakeFiles/pch.dir/cmake_pch.hxx.pch -Xclang -include -Xclang /opt/apache-doris-2.0.0-src/be/build_Release/CMakeFiles/pch.dir/cmake_pch.hxx -MD -MT src/agent/CMakeFiles/Agent.dir/heartbeat_server.cpp.o -MF src/agent/CMakeFiles/Agent.dir/heartbeat_server.cpp.o.d -o src/agent/CMakeFiles/Agent.dir/heartbeat_server.cpp.o -c /opt/apache-doris-2.0.0-src/be/src/agent/heartbeat_server.cpp
In file included from /opt/apache-doris-2.0.0-src/be/src/agent/heartbeat_server.cpp:31:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/storage_engine.h:40:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/calc_delete_bitmap_executor.h:30:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/rowset/rowset_writer.h:30:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/rowset/rowset_writer_context.h:24:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/tablet.h:49:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/rowset/rowset_reader.h:26:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/iterators.h:24:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/block_column_predicate.h:32:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/column_predicate.h:24:
/opt/apache-doris-2.0.0-src/be/src/olap/rowset/segment_v2/inverted_index_reader.h:183:10: error: virtual function 'visit' has a different return type ('void') than the function it overrides (which has return type 'int')
    void visit(int row_id, std::vector<uint8_t>& packed_value) override;
    ~~~~ ^
/opt/apache-doris-2.0.0-src/be/src/clucene/src/core/CLucene/util/bkd/bkd_reader.h:65:21: note: overridden virtual function is here
        virtual int visit(int docid, std::vector<uint8_t> &packedValue) = 0;
                ~~~ ^
1 error generated.
[3/860] Building CXX object src/agent/CMakeFiles/Agent.dir/task_worker_pool.cpp.o
FAILED: src/agent/CMakeFiles/Agent.dir/task_worker_pool.cpp.o
ccache /var/local/ldb-toolchain/bin/clang++ -DBOOST_STACKTRACE_USE_BACKTRACE -DHAVE_INTTYPES_H -DHAVE_NETINET_IN_H -DUSE_HADOOP_HDFS -I/opt/apache-doris-2.0.0-src/be/src/apache-orc/c++/include -I/opt/apache-doris-2.0.0-src/be/build_Release/src/apache-orc/c++/include -I/opt/apache-doris-2.0.0-src/be/build_Release/src/clucene/src/shared -I/opt/apache-doris-2.0.0-src/be/src/clucene/src/core -I/opt/apache-doris-2.0.0-src/be/src/clucene/src/shared -I/opt/apache-doris-2.0.0-src/be/src/clucene/src/contribs-lib -I/opt/apache-doris-2.0.0-src/be/src -I/opt/apache-doris-2.0.0-src/be/test -I/usr/lib/jvm/java-1.8.0/include -I/usr/lib/jvm/java-1.8.0/include/linux -isystem /opt/apache-doris-2.0.0-src/be/../gensrc/build -isystem /var/local/thirdparty/installed/include -isystem /var/local/thirdparty/installed/gperftools/include -O3 -DNDEBUG  -O3 -DNDEBUG   -D OS_LINUX -g -Wall -Wextra -Werror -pthread -fstrict-aliasing -fno-omit-frame-pointer -Wnon-virtual-dtor -Wno-unused-parameter -Wno-sign-compare -fcolor-diagnostics -Wpedantic -Wunused -Wunused-command-line-argument -Wunused-exception-parameter -Wunused-volatile-lvalue -Wunused-template -Wunused-member-function -Wunused-macros -Wno-vla-extension -D__STDC_FORMAT_MACROS -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX=1 -DBRPC_ENABLE_CPU_PROFILER -DS2_USE_GFLAGS -DS2_USE_GLOG -msse4.2 -mavx2 -DUSE_MEM_TRACKER -DUSE_JEMALLOC -DUSE_UNWIND -gdwarf-aranges -DENABLE_STACKTRACE -std=gnu++20 -Winvalid-pch -Xclang -include-pch -Xclang /opt/apache-doris-2.0.0-src/be/build_Release/CMakeFiles/pch.dir/cmake_pch.hxx.pch -Xclang -include -Xclang /opt/apache-doris-2.0.0-src/be/build_Release/CMakeFiles/pch.dir/cmake_pch.hxx -MD -MT src/agent/CMakeFiles/Agent.dir/task_worker_pool.cpp.o -MF src/agent/CMakeFiles/Agent.dir/task_worker_pool.cpp.o.d -o src/agent/CMakeFiles/Agent.dir/task_worker_pool.cpp.o -c /opt/apache-doris-2.0.0-src/be/src/agent/task_worker_pool.cpp
In file included from /opt/apache-doris-2.0.0-src/be/src/agent/task_worker_pool.cpp:18:
In file included from /opt/apache-doris-2.0.0-src/be/src/agent/task_worker_pool.h:35:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/tablet.h:49:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/rowset/rowset_reader.h:26:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/iterators.h:24:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/block_column_predicate.h:32:
In file included from /opt/apache-doris-2.0.0-src/be/src/olap/column_predicate.h:24:
/opt/apache-doris-2.0.0-src/be/src/olap/rowset/segment_v2/inverted_index_reader.h:183:10: error: virtual function 'visit' has a different return type ('void') than the function it overrides (which has return type 'int')
    void visit(int row_id, std::vector<uint8_t>& packed_value) override;
    ~~~~ ^
/opt/apache-doris-2.0.0-src/be/src/clucene/src/core/CLucene/util/bkd/bkd_reader.h:65:21: note: overridden virtual function is here
        virtual int visit(int docid, std::vector<uint8_t> &packedValue) = 0;
                ~~~ ^
1 error generated.
ninja: build stopped: subcommand failed.

**Comment by ixzc** [comment]
中文问题在ask.selectdb.com提个帖子
